components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java

```diff
@@ -1304,4 +1323,86 @@ protected void createLegacyUser(OAuthTokenReqMessageContext tokReqMsgCtx, Assert
         tokReqMsgCtx.setAuthorizedUser(user);
     }
 
+    /**
+     * Validate the signature against the certificate obtained from SAML Sign KeyStore which is defined under
+     * Security.SAMLSignKeyStore in carbon.xml.
+     *
+     * @param assertion
+     * @throws IdentityOAuth2Exception
+     */
+    protected void validateSignatureAgainstSAMLSignKeyStoreCertificate(Assertion assertion)
+            throws IdentityOAuth2Exception {
+
+        try {
+            X509Certificate x509Certificate = getCertificateFromSAMLSignKeyStore();
+            X509Credential x509Credential = new X509CredentialImpl(x509Certificate);
+            SignatureValidator signatureValidator = new SignatureValidator(x509Credential);
+            signatureValidator.validate(assertion.getSignature());
+        } catch (ValidationException e) {
+            throw new IdentityOAuth2Exception("Error while validating the signature from SAML sign keystore.", e);
```

Comment:
Can we put the assertion issuer or acs URL, to see which SP having the problem by looking at the logs

Reply:
added